libs/dataset_deployer.py

```diff
@@ -71,26 +71,23 @@ def flatten_dict(data: dict, level_separator: str = ".") -> dict:
 
 
 def write_nested_csv(
-    data: List[dict], output_path: pathlib.Path, keys_to_skip: Optional[List[str]] = None
+    data: List[dict], output_path: pathlib.Path, header: Optional[List[str]] = None,
 ):
     """Writes list of data as a nested csv.
 
     Args:
         data: list of data to write.
         output_path: Path of file to write to.
-        keys_to_skip: Keys to skip.  Keys can be flattened entries or top level keys.
-
+        header: Columns to output.  If specified, header will be in order of columns.
     """
-    keys_to_skip = keys_to_skip or []
-
     if not data:
         raise ValueError("Cannot upload a 0 length list.")
 
     first_row = data[0]
-    first_row = {key: value for key, value in first_row.items() if key not in keys_to_skip}
+    first_row = {key: value for key, value in first_row.items()}
 
-    header = flatten_dict(first_row).keys()
-    header = [column for column in header if column not in keys_to_skip]
+    all_columns = list(flatten_dict(first_row).keys())
```

Comment:
when will we find out that all_columns and header have diverged? a unit test would be nice, with an assert during production run is okayish, during QA is bad.

Reply:
I guess I'm leaning on the fact that this is not very likely - this would mean we've changed the names of the fields in the pydantic api field definitions.  In that case, we would be breaking the API for everyone using the JSON files - something I don't think we intend to do.

Reply:
What about a typo in `csv_column_ordering.py`?

Reply:
ah I think i'll add a test to make sure that these match properties in the matching schemas.